### PR TITLE
distro-base: python3 minimal poc

### PR DESCRIPTION
**Pull request by jaxesn**
*Issue #, if available:*

*Description of changes:*

@jonahjon needed a python 3.9 image for emissary in eks-a.  This adds a new python image with 3.9, for both al2 and al22 (for when we switch over).

There is a new pattern introduced here since there are a number of python versions still in support, even al2 has 2.7 3.7 and 3.8.  For now we are only building 3.9 since thats all we need, but we may introduce older ones just depending on need/asks.  There is only going to be one ecr repo for python and each image will tagged with its version, aka `3.9-latest` vs just `latest`.  The makefile has been changed to be `versioned_image` aware to handle these cases and the python dockerfile has the code to build 3.7 and 3.8, mainly as a POC/template for future builds that may require multiple versions.

I have built both platforms on both os versions locally and included the tracking files for reference.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: eks-distro-base/scripts/eks-d-common [hunk 1]
```diff
@@ -376,6 +376,15 @@ function build::common::clean_install() {
     local -r force=${4:-}
     local -r extract_dir="${5:-$NEWROOT}"
     
+
+    if [ ! $shallow ]; then
```
Comment (jaxesn) [inline eks-distro-base/scripts/eks-d-common:380]: Perf improvement for when we are just using yum to install a bunch of packages, this is significantly faster, esp thru qemu, than doing 1 by 1

File: eks-distro-base/scripts/eks-d-common [hunk 2]
```diff
@@ -393,72 +402,66 @@ function build::common::clean_install() {
 
     local -r rpms=(${packages// / })
     for rpm in "${rpms[@]}"; do
-        if [ ! $shallow ]; then
```
Comment (jaxesn) [inline eks-distro-base/scripts/eks-d-common:396]: This is all just a indentation change, no other changes.

File: eks-distro-base/Dockerfile.minimal-base-python [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+# Copy scripts in every variant since we do not rebuild the base
+# every time these scripts change. This ensures whenever a variant is
+# built it has the latest scripts in the builder
+COPY scripts/ /usr/bin
+
+# Google's Distroless has an experimental python3 version based on debian
+# The packages below are cross referenced based on the packages
+# included in distroless and the al2 python3 dependency tree
+# refernce: https://github.com/GoogleContainerTools/distroless/blob/main/experimental/python3/BUILD
```
Comment (markapruett) [inline eks-distro-base/Dockerfile.minimal-base-python:36]:
```suggestion
# reference: https://github.com/GoogleContainerTools/distroless/blob/main/experimental/python3/BUILD
```

File: eks-distro-base/Dockerfile.minimal-base-python [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    install_rpm python38-setuptools \
+        python38-pip \
+        python38 && \
+    # Symlink to just 3
+    ln -s python3.8 $NEWROOT/usr/bin/python3 && \
+    ln -s pip3.8 $NEWROOT/usr/bin/pip3 && \
+    cleanup "python3.8"
+
+
+FROM builder-python-base as builder-python3.9
```
[the remark this answers is not in the document]
Comment (jaxesn) [reply eks-distro-base/Dockerfile.minimal-base-python:95]: Yeah along with the makefile changes to support multiple versioned tags

**Comment by jaxesn** [comment]
Going to go ahead and merge this, but I welcome more reviews. These changes should only affect the new image being built, so risk is low. I want to kick off a build this morning since these things take hours.

**Pull request merged**